floodlight/io/catapultApi.py

```diff
@@ -0,0 +1,384 @@
+    if response.status_code == 200:
+        return response
+    else:
+        raise APIRequestError(f"API request failed with status code {response.status_code}")
+
+def getResponse_data_json_list(baseUrl: str, api_token: str, activity_id: str) -> List[Dict]:
+    """
+    Retrieves sensor data for each player involved in a specific activity.
+
+    Args:
+        baseUrl (str): The base URL for the API requests.
+        api_token (str): The API token for authentication.
+        activity_id (str): The unique identifier for the activity.
+
+    Returns:
+        List[Dict]: A list of JSON data responses for each player, containing sensor data.
+
+    Raises:
+        APIRequestError: If any API request fails with a status code other than 200.
+    """
+    headers = {
+        "accept": "application/json",
+        "authorization": f"Bearer {api_token}",
+        "nulls": "0"
+    }
+    allPlayersFromActivity_response = _getAllplayerFormActivity_response(baseUrl, api_token, activity_id)
+    allPlayersFromActivity_JSON = allPlayersFromActivity_response.json()
+    playerIDsInActivity_list = list(set(i["id"] for i in allPlayersFromActivity_JSON))
+    response_data_json_list = []
+
+    for i in playerIDsInActivity_list:
+        url = f'activities/{activity_id}/athletes/{i}/sensor'
+        response_data_json_i = requests.get(baseUrl + url, headers=headers)
+        if response_data_json_i.status_code == 200:
+            response_data_json_list.append(response_data_json_i.json()[0])
+            saveResponse(response_data_json_i, url)
+        else:
+            raise APIRequestError(f"API request failed with status code {response_data_json_i.status_code}")
+
+    return response_data_json_list
+
+def get_key_names_from_json(response_data_json_list: List[Dict]) -> set:
+    """
+    Extracts unique column names from a list of JSON data responses.
+
+    Args:
+        response_data_json_list (List[Dict]): List of JSON data responses.
+
+    Returns:
+        set: A set of unique column names.
+    """
+    recorded_columns = set()
+
+    for data_i in response_data_json_list:
+        recorded_columns.update(data_i.keys())
+        if "data" in data_i and isinstance(data_i["data"], list) and data_i["data"]:
+            recorded_columns.update(data_i["data"][0].keys())
+
+    return recorded_columns
+
+def _get_key_links(response_data_json_list: List[Dict]) -> Union[Dict[str, int], None]:
+    """
+    Generates a mapping of key names to their indices in the JSON data responses.
+
+    Args:
+        response_data_json_list (List[Dict]): List of JSON data responses.
+
+    Returns:
+        Union[Dict[str, int], None]: A dictionary mapping key names to their indices or None if critical keys are missing.
+    """
+    for data_i in response_data_json_list:
+        data_i['name'] = f"{data_i['athlete_first_name']} {data_i['athlete_last_name']}"
+
+    key_names = list(get_key_names_from_json(response_data_json_list))
+
+    mapping = {

```

Comment:
I don't think we need the mapping for every variable, but only those we are extracting. These would include every identifier, x, y, lat and long.

Reply:
Done